plugins/azure-sites-backend/src/service/router.ts

```diff
@@ -49,29 +59,63 @@ export async function createRouter(
     );
   });
   router.post(
-    '/:subscription/:resourceGroup/:name/start',
+    '/entity/:namespace/:kind/:name/sites/:subscription/:resourceGroup/:sitename/start',
     async (request, response) => {
-      const { subscription, resourceGroup, name } = request.params;
+      const { namespace, kind, name, subscription, resourceGroup, sitename } =
+        request.params;
+      const token = getBearerTokenFromAuthorizationHeader(
+        request.header('authorization'),
+      );
+      const resourceRef = stringifyEntityRef({ kind, namespace, name });
+      const decision = (
+        await permissions.authorize(
+          [{ permission: azureSitesActionPermission, resourceRef }],
+          {
+            token,
+          },
+        )
+      )[0];
```

Comment:
here we need to retrieve the sites and check that subscription, resourceGroup, sitename match with the ones referred by the `azure.com/microsoft-web-sites` annotation.
Otherwise, people might send namespace,kind,name of an entity they can access, together with subscription, resourceGroup, sitename of a site they can't access and the operation will be executed

Reply:
also, in case we want to simplify things a bit, we could encode subscription, resourceGroup, sitename in a single `siteId` string (with a separator between each field), so that you send back a single siteId to the backend for performing the operation.

Reply:
Implemented `azure.com/microsoft-web-sites` annotation comparison in backend, also simplified the API endpoint to `/azure-sites/:subscription/:resourceGroup/:name` same as before.

Reply:
nice looks good! Is there any reason why you are using a `RegExp` instead of simply checking whether `name === entity.metadata.annotation?.[AZURE_WEB_SITE_NAME_ANNOTATION]`?

Reply:
This Plugin offers exact/partial mapping of the `Azure Web Site` names in annotation, to match it both exact/partial I used `RegEx` 

https://github.com/backstage/backstage/blob/c4af1b249110bc6890b43a0babbc8f3d1104f598/plugins/azure-sites/README.md?plain=1#L27-L43

Reply:
cool, thanks for pointing out! 🙏

Reply:
one thing I need to point out is that the Regex matching might introduce some mismatch in the allowed resources.

In details:
the plugin retrieve the resources through the azure apis but then performs the authorization check manually through this Regex, which represents a different way for filtering the resources.

I wonder how big security issue this could be, given the fact that the goal of this PR is to protect the start/stop actions.  I feel like retrieving the list of the resources and performing a check whether an item with `{ subscription, resourceGroup, sitename }` exists is a much safer approach.

Reply:
I am not getting your question exactly.
The Frontend send stop/stop request with `{ subscription, resourceGroup, name }` along with the `entity reference`, before performing the action the Backend verify.
1. catalog permission - `azureSitesActionPermission`
2. whether the requested Azure Site `{ name }` is part of the catalog entity  via `AZURE_WEB_SITE_NAME_ANNOTATION` annotation with Regular expression.